configure.ac

```diff
@@ -26,38 +26,7 @@ AC_ARG_ENABLE(lz4, AS_HELP_STRING([--disable-lz4], [do not use lz4 compression])
 # Allow not linking to liblz4 even if it's present.
 AC_ARG_WITH([lz4], AS_HELP_STRING([--without-lz4], [never link to liblz4]))
 
-# Thanks to the OpenVPN configure.ac file for this part.
-# If this fails, we will do another test next.
-# We also add set LZ4_LIBS otherwise linker will not know about the lz4 library
-PKG_CHECK_MODULES(LZ4, [liblz4 >= 1.7.1], [have_lz4="yes"], [LZ4_LIBS="-llz4"])
-if test "${have_lz4}" != "yes" ; then
-    AC_CHECK_HEADERS([lz4.h],
-                     [have_lz4h="yes"],
-                     [])
-    if test "${have_lz4h}" = "yes" ; then
-        AC_MSG_CHECKING([additionally if system LZ4 version >= 1.7.1])
-        AC_COMPILE_IFELSE(
-        [AC_LANG_PROGRAM([[
-#include <lz4.h>
-                         ]],
-                         [[
-/* Version encoding: MMNNPP (Major miNor Patch) - see lz4.h for details */
-#if LZ4_VERSION_NUMBER < 10701L
-#error LZ4 is too old
-#endif
-                         ]]
-                        )],
-         [
-             AC_MSG_RESULT([ok])
-             have_lz4="yes"
-         ],
-         [
-             AC_MSG_RESULT([system LZ4 library is too old])
-             have_lz4="no"
-         ]
-        )
-    fi
-fi
+PKG_CHECK_MODULES(LZ4, [liblz4 >= 1.7.1], [have_lz4="yes"], [have_lz4="no"])
```

Comment:
`LZ4_LIBS` is not set anymore but it's used in `Makefile.am`. Or is there some magic that does it?

Reply:
`LZ4_LIBS` is set by the `PKG_CHECK_MODULES` macro, using what is reported by `pkg-config`.

The explicit set that was there in the code was there as a fallback, indeed it was set in the false branch. After the false branch was taken and `LZ4_LIBS` was set, the manual `if test "${have_lz4}" != "yes" ; then` fallback check would be executed, possibly leading to `have_lz4` being set to `yes`, and at that point `LZ4_LIBS` would remain set at the value set in the false branch of the `PKG_CHECK_MODULES` macro, and then be used in `Makefile.am`.

I'd say that the fallback `LZ4_LIBS` setting should have been performed inside the `if test "${have_lz4}" != "yes" ; then` fallback check, but the net effect is the same.